spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/configclient/ZookeeperConfigServerBootstrapper.java

```diff
@@ -53,10 +47,15 @@
 
 public class ZookeeperConfigServerBootstrapper implements BootstrapRegistryInitializer {
 
-	private static boolean isEnabled(Binder binder) {
-		return binder.bind(ConfigClientProperties.CONFIG_DISCOVERY_ENABLED, Boolean.class).orElse(false) &&
-				binder.bind(ConditionalOnZookeeperDiscoveryEnabled.PROPERTY, Boolean.class).orElse(true) &&
-				binder.bind("spring.cloud.discovery.enabled", Boolean.class).orElse(true);
+	private static PropertyResolver getPropertyResolver(BootstrapContext context) {
+		return context.getOrElse(PropertyResolver.class, new ConfigServerConfigDataLocationResolver.PropertyResolver(context.get(Binder.class), getBindHandler(context)));
```

Comment:
```suggestion
		return context.getOrElseSupply(PropertyResolver.class, () -> new ConfigServerConfigDataLocationResolver.PropertyResolver(
				context.get(Binder.class), context.getOrElse(BindHandler.class, null)));
```
The code underneath `getOrElse` seems to be creating a `Supplier` and passing it to `getOrElseSupply` anyway, but if we do that ourselves and call `getOrElseSupply` directly, we can avoid the unnecessary work associated with creating & GC-ing the fallback `PropertyResolver` instance when one is already registered in the context. (The potential savings are multiplied by the number of times this method gets called.)

In my suggestion, I also inlined `getBindHandler` according to my comment on that method.